utils/deploy/envoy.yaml

```diff
@@ -237,19 +238,21 @@ data:
                                               }
                                             },
                                             {
-                                              "selector": {
-                                                "selector": "source.address"
+                                              "expression": {
+                                                "key": "source.address",
+                                                "value": "source.address"
                                               }
                                             },
                                             {
-                                              "selector": {
-                                                "selector": "request.headers.My-Custom-Header-01"
+                                              "expression": {
+                                                "key": "request.headers['My-Custom-Header-01']",
+                                                "value": "request.headers['My-Custom-Header-01']"
                                               }
                                             },
                                             {
-                                              "selector": {
-                                                "selector": "metadata.filter_metadata.envoy\\.filters\\.http\\.header_to_metadata.user_id",
-                                                "key": "user_id"
+                                              "expression": {
+                                                "key": "user_id",
+                                                "value": "string(getHostProperty(['metadata', 'filter_metadata', 'envoy.filters.http.header_to_metadata', 'user_id']))"
```

Comment:
I wonder what would be required if the auth step is configured using gateway capabilities like [Security Policy]( https://gateway.envoyproxy.io/v0.6/design/security-policy/) (but not used for ext auth service, but `jwt` for instance.